x/bank/module.go

```diff
@@ -84,6 +85,9 @@ func (AppModuleBasic) GetQueryCmd() *cobra.Command {
 // RegisterInterfaces registers interfaces and implementations of the bank module.
 func (AppModuleBasic) RegisterInterfaces(registry codectypes.InterfaceRegistry) {
 	types.RegisterInterfaces(registry)
+
+	// Register legacy interfaces for migration scripts.
+	v040.RegisterInterfaces(registry)
```

Comment:
As long as we keep the migration scripts, we need to register legacy interfaces, which is kind of polluting our interfaceRegistry...

Maybe we could clone a separate interfaceRegistry just for migrations?
I created https://github.com/cosmos/cosmos-sdk/issues/9004 to discuss this.

Reply:
I don't think there is an easy way to solve it. There may be breaking changes related to proto files.